pennylane/spin/spin_hamiltonian.py

```diff
@@ -309,3 +309,308 @@ def fermi_hubbard(
+
+    spin = 2
+    hopping_ham = 0.0 * FermiWord({})
+    intersite_term = 0.0 * FermiWord({})
+    for edge in lattice.edges:
+        i, j, order = edge
+        if hopping.shape == (neighbour_order,):
+            hop = hopping[order]
+        else:
+            hop = hopping[i][j]
+
+        if intersite_coupling.shape == (neighbour_order,):
+            intersite = intersite_coupling[order]
+        else:
+            intersite = intersite_coupling[i][j]
+
+        for s in range(spin):
+            s1 = i * spin + s
+            s2 = j * spin + s
+            hopping_ham -= hop * (
+                FermiWord({(0, s1): "+", (1, s2): "-"}) + FermiWord({(0, s2): "+", (1, s1): "-"})
+            )
+        intersite_term += (
+            intersite
+            * (
+                FermiWord({(0, i * spin): "+", (1, i * spin): "-"})
+                + FermiWord({(0, i * spin + 1): "+", (1, i * spin + 1): "-"})
+            )
+            * (
+                FermiWord({(0, j * spin): "+", (1, j * spin): "-"})
+                + FermiWord({(0, j * spin + 1): "+", (1, j * spin + 1): "-"})
+            )
+        )
+
+    coulomb_term = 0.0 * FermiWord({})
+    if isinstance(coulomb, (int, float, complex)):
+        coulomb = math.ones(lattice.n_sites) * coulomb
+
+    for i in range(lattice.n_sites):
+        up_spin = i * spin
+        down_spin = i * spin + 1
+        coulomb_term += coulomb[i] * FermiWord(
+            {(0, up_spin): "+", (1, up_spin): "-", (2, down_spin): "+", (3, down_spin): "-"}
+        )
+
+    hamiltonian = hopping_ham + coulomb_term + intersite_term
+
+    if mapping not in ["jordan_wigner", "parity", "bravyi_kitaev"]:
+        raise ValueError(
+            f"The '{mapping}' transformation is not available."
+            f"Please set mapping to 'jordan_wigner', 'parity', or 'bravyi_kitaev'"
+        )
+    qubit_ham = qml.qchem.qubit_observable(hamiltonian, mapping=mapping)
+
+    return qubit_ham.simplify()
+
+
+def haldane(
+    lattice,
+    n_cells,
+    hopping1=1.0,
+    hopping2=1.0,
+    phi=1.0,
+    boundary_condition=False,
+    mapping="jordan_wigner",
+):
+    r"""Generates the Haldane Hamiltonian on a lattice defined in
+    `<https://arxiv.org/pdf/2211.13615>`__.
+
+    The Hamiltonian is represented as:
+
+    .. math::
+
+        \begin{align*}
+          \hat{H} & = -t_1 \sum_{\langle i,j \rangle}
+          (c_{i\sigma}^\dagger c_{j\sigma} + c_{j\sigma}^\dagger c_{i\sigma}) \\
+          & \quad - t_2 \sum_{\langle\langle i,j \rangle\rangle, \sigma}
+          \left( e^{i\phi_{ij}} c_{i\sigma}^\dagger c_{j\sigma} + e^{-i\phi_{ij}} c_{j\sigma}^\dagger c_{i\sigma} \right)
+        \end{align*}
+
+    where :math:`t_1` is the hopping term representing the hopping amplitude between neighbouring
+    sites, :math:`t_2` is the hopping amplitude between next nearest neighbours, :math:`\phi` is the phase
+    factor, :math:`\langle i,j \rangle`, :math:`\langle \langle i,j \rangle \rangle` represent the
```

Comment:
What does the phase factor signify?

Reply:
It represents a complex next nearest neighbor hopping. This term breaks time–reversal symmetry in the system, that's why there are positive and negative phase for hopping in two different directions.